corehq/apps/data_interfaces/static/data_interfaces/js/case_management.js

```diff
@@ -1,148 +1,217 @@
     };
 
-    self.updateCaseSelection = function (data, event) {
-        var $checkbox = $(event.currentTarget),
-            caseID = $checkbox.data('caseid'),
-            ownerID = $checkbox.data('owner'),
-            ownerType = $checkbox.data('ownertype');
-
-        var ind = self.selected_cases().indexOf(caseID),
-            $selectedRow = $checkbox.closest('tr');
-
-        if ($checkbox.is(':checked')) {
-            $selectedRow.addClass('active');
-            if (ind < 0) {
-                self.selected_cases.push(caseID);
-            }
-            self.selected_owner_types.push(ownerType);
-            self.selected_owners.push(ownerID);
-        } else {
-            $selectedRow.removeClass('active');
-            if (ind >= 0) {
-                self.selected_cases.splice(ind, 1);
+    ko.bindingHandlers.caseReassignmentForm = {
+        update: function(element, valueAccessor) {
+            var value = valueAccessor()();
+            var $element = $(element);
+            if (value.length > 0) {
+                $element.slideDown();
+            } else {
+                $element.find("[type='submit']").enableButton();
+                $element.slideUp();
             }
-            self.selected_owner_types.splice(self.selected_owner_types().indexOf(ownerType), 1);
-            self.selected_owners.splice(self.selected_owners().indexOf(ownerID), 1);
-        }
-    };
-
-    self.clearCaseSelection = function () {
-        self.selected_cases.removeAll();
+        },
     };
 
-    self.updateCaseOwners = function (form) {
-        var new_owner = $(form).find('#reassign_owner_select').val(),
-            $modal = $('#caseManagementStatusModal'),
-            owner_type = getOwnerType(new_owner);
+    $(function() {
+        var initialPageData = hqImport("hqwebapp/js/initial_page_data"),
+            interfaceSelector = '#data-interfaces-reassign-cases',
+            caseManagementModel = CaseManagement({
+                receiverUrl: initialPageData.reverse("receiver_secure_post"),
+                enddate: initialPageData.get("reassign_cases_enddate"),
+                webUserID: initialPageData.get("web_user_id"),
+                webUserName: initialPageData.get("web_username"),
+                form_name: gettext("Case Reassignment (via HQ)"),
+            });
+
+        // Apply bindings whenever report content is refreshed
+        $(document).on('ajaxSuccess', function(e, xhr, ajaxOptions) {
+            if (ajaxOptions.url.indexOf(initialPageData.get("js_options").asyncUrl) === -1) {
+                return;
+            }
 
-        if (new_owner.includes('__')) {
-            // groups and users have different number of characters before the id
-            // users are u__id and groups are sg__id
-            new_owner = new_owner.slice(new_owner.indexOf('__') + 2);
+            // Apply bindings to reassignment interface and select2
+            $(interfaceSelector).koApplyBindings(caseManagementModel);
```

Comment:
Just remembered one other question, possibly a blocker: does this need to be called before any `ajaxSuccess` events have fired? If yes, I didn't see where that happens.

Reply:
Yeah, I think there's a race condition here (if the ajax call finishes before this code runs, this will never execute). https://github.com/dimagi/commcare-hq/pull/20304/commits/8a5a672668195ff9daba08d15d72521524bcbaf8